src/utils/resources/vm/hooks/useVMI.ts

```diff
@@ -0,0 +1,29 @@
+import { VirtualMachineInstanceModelGroupVersionKind } from '@kubevirt-ui/kubevirt-api/console';
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseVMIForVM = (
+  vmName: string,
+  vmNamespace: string,
+) => {
+  vmi: V1VirtualMachineInstance;
+  vmiLoaded: boolean;
+  vmiLoadError: Error;
+};
+
+const useVMIForVM: UseVMIForVM = (vmName, vmNamespace) => {
+  const [vmi, vmiLoaded, vmiLoadError] = useK8sWatchResource<V1VirtualMachineInstance>({
+    groupVersionKind: VirtualMachineInstanceModelGroupVersionKind,
+    isList: false,
+    name: vmName,
+    namespace: vmNamespace,
+  });
+
+  return {
+    vmi: vmName === vmi?.metadata?.name && vmNamespace === vmi?.metadata?.namespace && vmi,
```

Comment:
only VMI i think is enough , as the filtering of vmname and vmnamespace is done in the request

Reply:
Fixed!